src/Riok.Mapperly/Helpers/EnumNamingStrategyHelper.cs

```diff
@@ -0,0 +1,40 @@
+using Microsoft.CodeAnalysis;
+using Riok.Mapperly.Abstractions;
+using Riok.Mapperly.Descriptors;
+
+namespace Riok.Mapperly.Helpers;
+
+public static class EnumNamingStrategyHelper
+{
+    public static IReadOnlyDictionary<IFieldSymbol, string> BuildCustomNameStrategyMappings(
```

Comment:
IMO this code could be simplified quite a bit by not creating a dictionary first. With the new approach of building the enum member mappings in the builder, each field is only processed once. Therefore there should be no need to collect to a dictionary first. My idea would be like this:
* replace `BuildCustomNameStrategyMappings` and `ConvertEnumValueNameToString` with an extension method `public static string GetName(this IFieldSymbol field, EnumNamingStrategy namingStrategy)`
* In `GetName` return the converted name (similar to the current `ConvertEnumValueNameToString`)

In `StringToEnumMappingBuilder` it could then look like this:
```csharp
            ...
            if (namingStrategy == EnumNamingStrategy.MemberName)
            {
                // nameof(source.Value1)
                yield return new EnumMemberMapping(NameOf(targetSyntax), targetSyntax);
                continue;
            }

            // "value1"
            var name = targetField.GetName(namingStrategy);
            yield return new EnumMemberMapping(StringLiteral(name), targetSyntax);
           ....
```
and in `EnumToStringMappingBuilder`
```csharp
            ...
            var name = sourceField.GetName(namingStrategy);
            if (string.Equals(name, fallbackStringValue, StringComparison.Ordinal))
                continue;

            if (namingStrategy != EnumNamingStrategy.MemberName)
            {
                // "value1"
                yield return new EnumMemberMapping(sourceSyntax, StringLiteral(name));
                continue;
            }

            // nameof(source.Value1)
            yield return new EnumMemberMapping(sourceSyntax, NameOf(sourceSyntax));
            ...
```
WDYT?

Reply:
Agreed, simplified the code as suggested by removing the dictionary and using the extension method.